batchflow/models/torch/blocks/attention.py

```diff
@@ -141,6 +143,44 @@ def __repr__(self):
         return layer_desc
 
 
+class EfficientMultiHeadAttention(nn.Module):
+    """ Attention layer, popularized by transformer architectures.
+    Efficient in a sense of reducing the number of sequence elements `ratio^2` times.
+    Reduction is implemented as a convolution, and attention is implemented by a native `PyTorch` layer.
+
+    Parameters
+    ----------
+    ratio : int
+        Spatial reduction ratio. As this is applied across both spatial dimensions,
+        the actual reduction in number of sequence elements is `ratio` squared.
+    num_heads : int
+        Number of parallel attention heads. Must be a divisor of `input` number of channels.
+    """
+    def __init__(self, inputs=None, ratio=4, num_heads=8):
```

Comment:
why without layernorm as in original EfficientMultiHeadAttention?

Reply:
I decoupled the layernorm, as well as skip conection, from the actual operation. In the named blocks, I am using the `RnS+` layout to include the layernorm/skip: it seems more explicit to me